packages/utils/src/signatureUtils.ts

```diff
@@ -129,6 +130,20 @@ export function parseSignatureWithoutPrefix(
   throw new Error(`Unable to parse signature (expected signer ${signer})`)
 }
 
+export function guessEIP712TypedDataSigner(typedData: EIP712TypedData, signature: string): string {
+  const dataBuff = generateTypedDataHash(typedData)
```

Comment:
The issue is that then, everything done between now and when we fix it is incorrect... 

How much stuff would merging your PR break?

Reply:
That's the question -- I wasn't able to track down any usages of the `signTypedData` so I'm not sure what will break. That was an open question I had as well in the PR. 

At the moment I updated it, fixed the tests and also figured out the Ledger has a specialised method for EIP712 and used that in the signer as well. So I think it's in a good spot, but because of the nature of the beast the tests are a bit brittle / irrelevant because basically the tests for the Azure HSM/AWS HSM/Ledger signers mock the actual signing logic so it's basically just testing that the wrappers behave correctly. I guess we would still need to manually verify that signing happens correctly with the respective key managers - but, at least in the HSM case I think it should be ok because we're constructing the message hash in javascript-land, which leaves the Ledger where we're trusting their internal library's implementation.

Anyway coming back to the actual question I'm not sure what would break now by merging that, it looks like, at least in `celo-monorepo` nothing is actually consuming the `signTypedData` logic - except for the MetaTransactionWallet which expects the correct version, but I'm unsure if there are other consumers of ContractKit that will be affected.

One option maybe worth considering would be to expose both signing schemes for the sake of backwards compatibility but I'm worried that it would compound the confusion.

Reply:
+1 on compounding confusion. I'm also under the suspicion no one is using the current implementation. So let's merge.